x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptMappedFieldType.java

```diff
@@ -76,6 +78,22 @@ public abstract Query rangeQuery(
         ZoneId timeZone,
         DateMathParser parser,
         QueryShardContext context
+    ) {
+        if (relation == ShapeRelation.DISJOINT) {
+            String message = "Field [%s] of type [%s] with runtime type [%s] does not support DISJOINT ranges";
+            throw new IllegalArgumentException(String.format(Locale.ROOT, message, name(), typeName(), runtimeType()));
```

Comment:
I went with this rather funny way of writing the message because it made the formatter happy. Most of the other options I tried caused the formatter to split the message over so many lines it wasn't readable. I don't generally like String.format for these but here it looks best, I think.